Source/VMware.vSphereDSC/Classes/InventoryBaseDSC.ps1

```diff
@@ -118,13 +120,14 @@ class InventoryBaseDSC : BaseDSC {
             $childEntities = Get-View -Server $this.Connection -Id $foundPathItem.ChildEntity
         }
 
-        try {
-            $datacenterLocation = Get-Inventory -Server $this.Connection -Id $foundPathItem.MoRef
-            return Get-Datacenter -Server $this.Connection -Name $datacenterName -Location $datacenterLocation -ErrorAction Stop | Where-Object { $._ParentFolderId -eq $datacenterLocation.Id }
-        }
-        catch {
-            throw "Datacenter with name $datacenterName was not found. For more inforamtion: $($_.Exception.Message)"
+        $datacenterLocation = Get-Inventory -Server $this.Connection -Id $foundPathItem.MoRef
+        $foundDatacenter = Get-Datacenter -Server $this.Connection -Name $datacenterName -Location $datacenterLocation | Where-Object { $_.ParentFolderId -eq $datacenterLocation.Id }
```

Comment:
Same as above here, how do you deal with the error when datacenter with name is not found?